test/new-e2e/tests/agent-metrics-logs/log-agent/linux-log/integrations/logs_test.go

```diff
@@ -7,40 +7,145 @@ package integrationslogs
 
 import (
 	_ "embed"
+	"regexp"
+	"strconv"
+	"strings"
 	"testing"
+	"time"
+
+	"gopkg.in/yaml.v3"
 
 	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
 	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
 	awshost "github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments/aws/host"
 	"github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-metrics-logs/log-agent/utils"
 	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/stretchr/testify/assert"
 )
 
 type IntegrationsLogsSuite struct {
 	e2e.BaseSuite[environments.Host]
 }
 
-//go:embed fixtures/tenLogs.py
-var writeTenLogsCheck string
+//go:embed fixtures/integration.py
+var customIntegration string
+
+type Config struct {
+	InitConfig interface{}  `yaml:"init_config"`
+	Instances  []Instance   `yaml:"instances"`
+	Logs       []LogsConfig `yaml:"logs"`
+}
+
+type Instance struct {
+	LogMessage      string `yaml:"log_message"`
+	UniqueMessage   bool   `yaml:"unique_message"`
+	LogSize         int    `yaml:"log_size"`
+	LogCount        int    `yaml:"log_count"`
+	IntegrationTags string `yaml:"integration_tags"`
+}
 
-//go:embed fixtures/tenLogs.yaml
-var writeTenLogsConfig string
+type LogsConfig struct {
+	Type    string `yaml:"type"`
+	Source  string `yaml:"source"`
+	Service string `yaml:"service"`
+}
 
 // TestLinuxFakeIntakeSuite
 func TestIntegrationsLogsSuite(t *testing.T) {
 	suiteParams := []e2e.SuiteOption{
 		e2e.WithProvisioner(awshost.Provisioner(awshost.WithAgentOptions(
 			agentparams.WithLogs(),
-			// set the integration log file max size to 1MB
-			agentparams.WithAgentConfig("logs_config.integrations_logs_files_max_size: 1"),
-			agentparams.WithFile("/etc/datadog-agent/checks.d/writeTenLogs.py", writeTenLogsCheck, true),
-			agentparams.WithFile("/etc/datadog-agent/conf.d/writeTenLogs.yaml", writeTenLogsConfig, true))))}
+			agentparams.WithAgentConfig("logs_config.integrations_logs_files_max_size: 1"))))}
+
+	suiteParams = append(suiteParams, e2e.WithDevMode())
 
 	e2e.Run(t, &IntegrationsLogsSuite{}, suiteParams...)
 }
 
 // TestWriteTenLogsCheck ensures a check that logs are written to the file ten
 // logs at a time
 func (v *IntegrationsLogsSuite) TestWriteTenLogsCheck() {
-	utils.CheckLogsExpected(v.T(), v.Env().FakeIntake, "ten_logs_service", "Custom log message", []string{"env:dev", "bar:foo"})
+	tags := []string{"foo:bar", "env:dev"}
+	yamlData, err := generateYaml("Custom log message", false, 1, 10, tags, "logs_from_integrations_source", "logs_from_integrations_service")
+	assert.NoError(v.T(), err)
+
+	v.UpdateEnv(awshost.Provisioner(awshost.WithAgentOptions(
+		agentparams.WithLogs(),
+		agentparams.WithFile("/etc/datadog-agent/conf.d/writeTenLogs.yaml", string(yamlData), true),
+		agentparams.WithFile("/etc/datadog-agent/checks.d/writeTenLogs.py", customIntegration, true))))
+
+	utils.CheckLogsExpected(v.T(), v.Env().FakeIntake, "logs_from_integrations_service", "", tags)
```

Comment:
No check to see if you got at least ten logs.

Reply:
Changed!